azkaban-common/src/main/java/azkaban/database/AbstractJdbcLoader.java

```diff
@@ -63,8 +71,14 @@ public AbstractJdbcLoader(Props props) {
     dataSource = DataSourceUtils.getDataSource(props);
   }
 
+  @Override
+  public synchronized HashSet<EventListener> getListeners() {
+    return MultitonListenerSet.getInstance(MultitonListenerSet.ListenerType.COMMON, CommonMetrics.INSTANCE).getListeners();
+  }
+
   protected Connection getDBConnection(boolean autoCommit) throws IOException {
     Connection connection = null;
+    fireEventListeners(Event.create(this, Event.Type.DB_CONNECTION));
```

Comment:
Does getDBConnection always create connections? If yes, should we rename the method to something like `createDBConnnection` ?